### PR TITLE
Add reflowCursorLine option

### PR DESCRIPTION
**Pull request by jtbandes**
Resolves https://github.com/xtermjs/xterm.js/issues/5213

cc @Tyriar @jerch – per our discussions on https://github.com/xtermjs/xterm.js/issues/5213 this is adding an option to enable wrapping of the line with the cursor. This is my first time trying to modify xterm so please let me know if you have any suggestions for improvements.

I also updated the demo app with a checkbox to disable the real terminal, so it was easier for me to enable the fake one for testing (because a real shell will automatically wrap its prompt line as discussed in https://github.com/xtermjs/xterm.js/issues/5213#issuecomment-2488439326).

It looks like this results in the cursor being in a different place than where it started. If you have any ideas for how to resolve this, let me know. Or maybe this behavior is expected?

https://github.com/user-attachments/assets/93b45072-b64b-4a01-a50f-51cc67822583

### CODE REVIEW

File: src/common/buffer/BufferReflow.ts [hunk 1]
```diff
@@ -20,8 +20,9 @@ export interface INewLayoutResult {
  * @param newCols The columns after resize.
  * @param bufferAbsoluteY The absolute y position of the cursor (baseY + cursorY).
  * @param nullCell The cell data to use when filling in empty cells.
+ * @param reflowCursorLine Whether to reflow the line containing the cursor.
  */
-export function reflowLargerGetLinesToRemove(lines: CircularList<IBufferLine>, oldCols: number, newCols: number, bufferAbsoluteY: number, nullCell: ICellData): number[] {
+export function reflowLargerGetLinesToRemove(lines: CircularList<IBufferLine>, oldCols: number, newCols: number, bufferAbsoluteY: number, nullCell: ICellData,reflowCursorLine: boolean): number[] {
```
Comment (Tyriar) [inline src/common/buffer/BufferReflow.ts:25]:
```suggestion
export function reflowLargerGetLinesToRemove(lines: CircularList<IBufferLine>, oldCols: number, newCols: number, bufferAbsoluteY: number, nullCell: ICellData, reflowCursorLine: boolean): number[] {
```

File: src/common/buffer/Buffer.ts [hunk 2]
```diff
@@ -315,7 +315,7 @@ export class Buffer implements IBuffer {
   }
 
   private _reflowLarger(newCols: number, newRows: number): void {
-    const toRemove: number[] = reflowLargerGetLinesToRemove(this.lines, this._cols, newCols, this.ybase + this.y, this.getNullCell(DEFAULT_ATTR_DATA));
+    const toRemove: number[] = reflowLargerGetLinesToRemove(this.lines, this._cols, newCols, this.ybase + this.y, this.getNullCell(DEFAULT_ATTR_DATA), this._optionsService.rawOptions.reflowCursorLine);
```
Comment (Tyriar) [inline src/common/buffer/Buffer.ts:318]:
```suggestion
    const reflowCursorLine = this._optionsService.rawOptions.reflowCursorLine;
    const toRemove: number[] = reflowLargerGetLinesToRemove(this.lines, this._cols, newCols, this.ybase + this.y, this.getNullCell(DEFAULT_ATTR_DATA), reflowCursorLine);
```

**Review by Tyriar — APPROVED** [review]

**Comment by Tyriar** [comment]
Looks great, just need to test it 👍

**Comment by jtbandes** [comment]
@Tyriar thanks! Any idea when this will be published in a new beta release?

**Comment by jtbandes** [comment]
Actually it looks like it tried to publish, but the publish job failed: https://github.com/xtermjs/xterm.js/actions/runs/12298607666/job/34322412357

**Pull request merged**